### ISSUE TITLE
Allow Grid RowDefinitions and ColumnDefinitions defined in attributes

### ISSUE THREAD

**Issue by wieslawsoltes** [Microsoft/xaml-standard#47]
Allow using attributes to define `Grid` `RowDefinitions` and `ColumnDefinitions`
```XAML
   <Grid RowDefinitions="50,*" ColumnDefinitions="Auto,*" Background="Ivory">
      <Button Grid.Row="1" Grid.Column="0" ... />
      ...
   </Grid>
```
as alternative to proposed more verbose syntax:
```XAML
   <Grid Background="Ivory">
      <Grid.RowDefinitions>
         <RowDefinition Height="50"/>
         <RowDefinition Height="*"/>
      </Grid.RowDefinitions>
      <Grid.ColumnDefinitions>
         <ColumnDefinition Width="Auto" />
         <ColumnDefinition Width="*" />
      </Grid.ColumnDefinitions>
      <Button Grid.Row="1" Grid.Column="0" ... />
      ...
   </Grid>
```
Reference: https://github.com/Microsoft/xaml-standard/blob/staging/docs/v1draft.md#5-grid

**Comment by Odonno** [comment]
@wieslawsoltes Clearly, it is leaner to use `Grid.Row` and `Grid.Column`.

**Comment by billreiss** [comment]
Curious why anyone would downvote this

**Comment by Odonno** [comment]
@billreiss Yeah, it would be much better to argument against this proposition instead of just pushing a button.

**Comment by Mike-EEE** [comment]
> Curious why anyone would downvote this

Do you even internet, bro? 😆

**Comment by BradChase2011** [comment]
I was wondering and I am not sure the best place to ask this because it seems RowDefinition and ColumnDefinition are already put into the standard, but why are we still using RowDefinition and ColumnDefinition?  Why not create one type.  Here is an example of what I have in my custom rendering xaml:

<Grid.RowDefinitions>
         <Section Length="50"/>
         <Section Length="*"/>
      </Grid.RowDefinitions>
      <Grid.ColumnDefinitions>
         <Section Length="Auto" />
         <Section Length="*" />
      </Grid.ColumnDefinitions>

it makes it MUCH easier when building out new screens and especially cleaning up code with copy and pasting.

**Comment by Odonno** [comment]
@BradChase2011 It seems interesting.

**Comment by danwalmsley** [comment]
This new way of defining rows and columns is really great! 

```
RowDefinitions="50,*" ColumnDefinitions="Auto,*" 
```

Any time I help introduce someone to UWP, WPF or other XAML framework (that is not avalonia which already has this)

I generally get a negative reaction!
It has to be one of the things people hate the most about XAML 

```
      <Grid.RowDefinitions>
         <RowDefinition Height="50"/>
         <RowDefinition Height="*"/>
      </Grid.RowDefinitions>
      <Grid.ColumnDefinitions>
         <ColumnDefinition Width="Auto" />
         <ColumnDefinition Width="*" />
      </Grid.ColumnDefinitions>
```

**Comment by GeraudFabien** [comment]
@danwalmsley I don't thinks it making the code harder to maintain. In VS you can collapse so it did not disturb you. More in **my** experience most of the time you will have two/three style of column&row were you just have to set the style. This way i set all the MinHeight and MaxHeight and Height (or width for column) one time. I didn't like the proposal because i prefer somethinks like [Rows/Columns]Definitions="FirstStyle,.." Where FirstStyle can be a number or any static ressource. It would enable what is ask here. But also the scenario i talk berore.

**Comment by Odonno** [comment]
Oh sorry, I only updated for `Grid.Row` and `Grid.Column`. I did not see the simplified definition of rows and columns.

At first sight, it seems simple and but like @GeraudFabien said there is always one time I will use MinHeight/MaxHeight properties. So, not sure the definition of rows/columns should be integrated, at least  in the v1.

**Comment by birbilis** [comment]
it does make XAML ugly and hard to read (too much info) indeed - wouldn't mind to also have concise ways, the thing is how one can make them generic and not control specific when possible. So that you can use bindings etc. on them

**Comment by Jehoel** [comment]
@wieslawsoltes Why did you close this issue? It's still a valid problem.

**Comment by wieslawsoltes** [comment]
> @wieslawsoltes Why did you close this issue? It's still a valid problem.

https://github.com/microsoft/xaml-standard#updates
